java/gateway-core/src/main/java/com/avaulta/gateway/rules/PathTemplateUtils.java

```diff
@@ -0,0 +1,50 @@
+package com.avaulta.gateway.rules;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PathTemplateUtils {
+
+
+    //TODO: improve this; some special chars outside of {} are not accounted for
+    private static final String SPECIAL_CHAR_CLASS = "[\\.\\^\\$\\<\\>\\*\\+\\[\\]\\(\\)\\+\\-\\=\\?\\!]";
+
+    private static final String REGEX_ALPHANUMERIC_PATH_PARAM = "\\{([A-Za-z][A-Za-z0-9]*)\\}";
```

Comment:
just FYI, not sure if more readable or not, but these Posix "shortcuts" exists in java regexp
`\\{(\p{Alpha}\p{Alnum}*)\\}`

Reply:
yeah, imho not sure is more readable.